packages/messageport/tizen/src/messageport.cc

```diff
@@ -185,21 +171,31 @@ MessagePortResult MessagePortManager::Send(std::string& remote_app_id,
   }
   bundle_free(bundle);
 
-  return CreateResult(ret);
+  if (ret != MESSAGE_PORT_ERROR_NONE) {
+    return MessagePortError(ret);
+  }
+
+  return std::nullopt;
 }
 
-MessagePortResult MessagePortManager::Send(std::string& remote_app_id,
-                                           std::string& port_name,
-                                           flutter::EncodableValue& message,
-                                           bool is_trusted, int local_port) {
-  LOG_DEBUG("Send (%s, %s), port: %d, trusted: %s", remote_app_id.c_str(),
-            port_name.c_str(), local_port, is_trusted ? "yes" : "no");
-  bundle* bundle = nullptr;
-  MessagePortResult result = PrepareBundle(message, bundle);
-  if (!result) {
-    return result;
-  }
-  int ret;
+std::optional<MessagePortError> MessagePort::Send(
+    std::string& remote_app_id, std::string& port_name,
+    std::unique_ptr<std::vector<uint8_t>> encoded_message, bool is_trusted,
+    const std::string& local_port_name, bool local_port_is_trusted) {
```

Comment:
You may specify `remote` and `local` appropriately in the argument names (e.g. `remote_port_name`, `trusted_remote_port`) to avoid ambiguity.

As there are too many arguments to this method, a better way to refactor the code could be splitting the whole class into two (`LocalPort` and `RemotePort`), and let each class has `name` and `is_trusted` as members (as well as other relevant methods). In this way the code will become more object oriented and the singleton pattern will no longer be necessary.